metricbeat/helper/sql/sql.go

```diff
@@ -0,0 +1,187 @@
+	// https://stackoverflow.com/questions/23507531/is-golangs-sql-package-incapable-of-ad-hoc-exploratory-queries/23507765#23507765
+	cols, err := rows.Columns()
+	if err != nil {
+		return nil, errors.Wrap(err, "error getting columns")
+	}
+
+	for k, v := range cols {
+		cols[k] = strings.ToLower(v)
+	}
+
+	vals := make([]interface{}, len(cols))
+	for i := 0; i < len(cols); i++ {
+		vals[i] = new(interface{})
+	}
+
+	rr := make([]common.MapStr, 0)
+	for rows.Next() {
+		err = rows.Scan(vals...)
+		if err != nil {
+			d.logger.Debug(errors.Wrap(err, "error trying to scan rows"))
+			continue
+		}
+
+		r := common.MapStr{}
+
+		for i, c := range cols {
+			value := getValue(vals[i].(*interface{}))
+			r.Put(c, value)
+		}
+
+		rr = append(rr, r)
+	}
+
+	if err = rows.Err(); err != nil {
+		d.logger.Debug(errors.Wrap(err, "error trying to read rows"))
+	}
+
+	return rr, nil
+}
+
+
+// fetchTableMode scan the rows and publishes the event for querys that return the response in a table format.
+func (d *Db) FetchVariableMode(ctx context.Context, q string) (common.MapStr, error) {
+	rows, err := d.QueryContext(ctx, q)
+	if err != nil {
+		return nil, err
+	}
+	return d.fetchVariableMode(rows)
+}
+
+// fetchVariableMode scan the rows and publishes the event for querys that return the response in a key/value format.
+func (d *Db) fetchVariableMode(rows sqlRow) (common.MapStr, error) {
+	data := common.MapStr{}
+
+	for rows.Next() {
+		var key string
+		var val interface{}
+		err := rows.Scan(&key, &val)
+		if err != nil {
+			d.logger.Debug(errors.Wrap(err, "error trying to scan rows"))
+			continue
+		}
+
+		key = strings.ToLower(key)
+		data[key] = val
+	}
+
+	if err := rows.Err(); err != nil {
+		d.logger.Debug(errors.Wrap(err, "error trying to read rows"))
+	}
+
+	r := common.MapStr{}
+
+	for key, value := range data {
+		value := getValue(&value)
+		r.Put(key, value)
+	}
+
+	return r, nil
+}
+
+func ToDotKeys(ms common.MapStr) common.MapStr {
+	dotMap := common.MapStr{}
+	for k, v := range ms {
+		dotMap.Put(strings.Replace(k, "_", ".", -1), v)
+	}
+
+	return dotMap
+}
+
+func getValue(pval *interface{}) interface{} {
+	switch v := (*pval).(type) {
+	case nil, bool:
+		return v
+	case []byte:
+		s := string(v)
+		num, err := strconv.ParseFloat(s, 64)
+		if err == nil {
+			return num
+		}
+		return s
+	case time.Time:
+		return v.Format(time.RFC3339Nano)
+	case []interface{}:
+		return v
+	default:
+		s := fmt.Sprint(v)
+		num, err := strconv.ParseFloat(s, 64)
+		if err == nil {
+			return num
+		}
+		return s
+	}
+}
+
+// switchDriverName switches between driver name and a pretty name for a driver. For example, 'oracle' driver is called
+// 'godror' so this detail implementation must be hidden to the user, that should only choose and see 'oracle' as driver
+func switchDriverName(d string) string {
+	switch d {
+	case "oracle":
+		return "godror"
+	case "cockroachdb":
+		return "postgres"
+	case "cockroach":
+		return "postgres"
+	case "postgresql":
```

Comment:
`mysql` driver?

Reply:
`mysql` do not require a "translation" as everybody call it `mysql`. Example: The fact that the library of oracle is called `godror` is an implementation detail that Beats shouldn't expose to users. Users will expect to write `oracle` as the driver name. Same with CockroachDB which uses `postgres` driver or `postgresql ~= postgres`.

It can be very annoying to write `postgresql` as the driver name and receive a `driver not recognized` because the actual driver name is `postgres` (without `ql`)

Reply:
Gotcha, thanks!